File: UEPlugin/FCScript/Content/Wrap/FCTest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

//typdef DelegateEvent  FOnButtonClickedEvent
//typdef MulticastDelegate  FOnHttpResponseMessage

public class UFCTest : UObject
{
	public float GetHP(){return 0;}
	public void SetIDList(TArray<int> IDs){}
	public bool GetIDList(out TArray<int> IDs){}
	public void SetNameList(TArray<StringA> Names){}
	public void NotifyObject(int nType, float x, float y, float z){}
	public static int NotifyAll(int nType, Vector3 Pos){}
	public void HttpNotify(StringA MessageContent, bool bWasSuccessful){}
	public void CallClicked(){}

    public TArray<int> IDList { get; set; }
    public TArray<StringA> NameList { get; set; }

    public UFCTest NextPtr { get;  set; }
	public int ID { get;  set; }
	public float HP { get;  set; }
	public List<int> aID { get;  set; }
	public Vector3 Pos { get;  set; }
	public MulticastDelegateEvent OnClicked { get;  set; }
	public DelegateEvent OnResponseMessage { get;  set; }
};

File: UEPlugin/Plugins/FCScript/Source/FCScript/Demo/FCTest.cpp
#include "FCTest.h"

float UFCTest::GetHP() const
{
    return HP;
}

void UFCTest::SetIDList(const TArray<int32>& IDs)
{
    IDList = IDs;
}

bool UFCTest::GetIDList(TArray<int32>& IDs)
{
    IDs = IDList;
    return true;
}

void UFCTest::SetIDMap(const TMap<int32, int32>& IDs)
{
    IDMap = IDs;
}

void UFCTest::GetIDS(TArray<int32>& OutIDS, TMap<int32, int32>& OutMap)
{
    OutIDS = IDList;
    OutMap = IDMap;
}

void UFCTest::SetIDSet(const TSet<int32>& InIDSet)
{
    IDSet = InIDSet;
}

void UFCTest::GetIDSet(TSet<int32>& OutIDSet)
{
    OutIDSet = IDSet;
}

void UFCTest::SetNameList(const TArray<FString>& Names)
{
    NameList = Names;
}

int UFCTest::NotifyAll(int nType, const FVector& Pos)
{
    return 100 + nType;
}

void UFCTest::HttpNotify(const FString& MessageContent, bool bWasSuccessful)
{
    OnResponseMessage.ExecuteIfBound(MessageContent, bWasSuccessful);
}

void UFCTest::CallClicked()
{
    OnClicked.Broadcast();
}

void UFCTest::CallDoubleClicked()
{
    OnDoubleClicked.Broadcast();
}
